ftw/publisher/sender/persistence.py

```diff
@@ -253,29 +254,67 @@ def __init__(self, context):
         """
         self.context = aq_inner(context.portal_url.getPortalObject())
         self.annotations = IAnnotations(self.context)
+        self.logger = getLogger()
 
     security.declarePrivate('_get_jobs_queue')
     def _get_jobs_queue(self):
         if 'publisher-queue' not in self.annotations:
             self.annotations['publisher-queue'] = zc.queue.Queue()
         return self.annotations['publisher-queue']
 
+    security.declarePrivate('_get_worker_queue')
+    def _get_worker_queue(self):
+        if 'worker-queue' not in self.annotations:
+            self.annotations['worker-queue'] = zc.queue.Queue()
+        return self.annotations['worker-queue']
+
+    security.declarePrivate('get_worker_queue')
+    def get_worker_queue(self):
+        return self._get_worker_queue()
+
+    security.declarePrivate('move_to_worker_queue')
+    def move_to_worker_queue(self):
+        job_queue = self._get_jobs_queue()
+        job_counter = 0
```

Comment:
`job_counter` seems unnecessary.
Instead, I'd early return when nothing to do in order to not commit since we have not really changed the queue:
```python
if not len(job_queue):
    return
```